### PR TITLE
PowerFlex block storage deployment on Azure

### PR DESCRIPTION
**Pull request by rfsonders**
Add module/examples for Powerflex block storage deployment on Azure.
Add guide on multi steps for the Powerflex block storage deployment on Azure.

### CODE REVIEW

File: examples/azure_pfmp/terraform.tfvars [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+is_multi_az             = false
+existing_resource_group = null # If null, a new RG will be created, else provide an existing "RG name"
+location                = "westus3"
+prefix                  = "sonders-tf-test"
+### data disk
+### the following value won't take effect on optimized_v1 and optimized_v2 deployment
+### it will always be:
+###     data_disk_count = 4 data_disk_size_gb=1966.08 for optimized_v1
+###     data_disk_count = 8 data_disk_size_gb=1966.08 for optimized_v2
+data_disk_count   = 2
```
Comment (vangork) [inline examples/azure_pfmp/terraform.tfvars:33]: Should be kept 20 for default.

File: examples/azure_pfmp/terraform.tfvars [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# */
+
+### cluster
+cluster_node_count      = 5 # Minimum of 5. If is_multi_az = true, then minimum of 6
+enable_bastion          = true
+enable_jumphost         = true
+enable_sql_workload_vm  = true # If enabled, will deploy Standard_D4d_v5
```
Comment (vangork) [inline examples/azure_pfmp/terraform.tfvars:22]: Do we need to enable sql vm by default?

File: modules/azure_pfmp/main.tf [hunk 3]
```diff
@@ -0,0 +1,532 @@
...
+    managed_disk_type    = "Premium_LRS"
+   # storage_account_type = "Premium_LRS"
+  }
+  
+  os_profile {
+    computer_name  = "${var.prefix}-sql-vm"
+    admin_username = var.login_credential.username
+    admin_password = var.login_credential.password
+  }
+  os_profile_windows_config {
```
Comment (vangork) [inline modules/azure_pfmp/main.tf:221]: If no additional windows config needed, delete this block

File: examples/azure_pfmp/terraform.tfvars (hunk 1 above)
Comment (rfsonders) [reply examples/azure_pfmp/terraform.tfvars:33]: Correct and valid.

File: examples/azure_pfmp/terraform.tfvars (hunk 2 above)
Comment (rfsonders) [reply examples/azure_pfmp/terraform.tfvars:22]: Good thought. The SQL VM should be false by default.

File: modules/azure_pfmp/main.tf (hunk 3 above)
Comment (rfsonders) [reply modules/azure_pfmp/main.tf:221]: I saw an error during deployment that referenced a need for this block. 
Maybe provition_vm_agent = true is needed here.  
https://github.com/hashicorp/terraform-provider-azurerm/issues/9219

File: modules/azure_pfmp/providers.tf
Comment (doriac11) [inline modules/azure_pfmp/providers.tf:1]: Shouldnt these providers be also include in the example providers.tf?

Comment (vangork) [reply modules/azure_pfmp/providers.tf:1]: no need, it will auto import when init the module.

**Review by doriac11 — APPROVED** [review]

**Comment by baoy1** [comment]
Hi @vangork , not sure if "Rebase and Merge" is appliable, would you squash the 37 commits to one?

**Review by forrestxia — APPROVED** [review]

**Pull request merged**